### PR TITLE
Rework click events to be less hacky

### PR DESCRIPTION
**Pull request by bensku**
### Description
Click events are a long standing pain point in Skript. They're not reliable enough across all supported Minecraft versions. Some use cases, like clicks on entities, are almost univerally broken.

This pull request abandons lookup tables for click event deduplication. Instead, scripts will simply never receive more than one click event per tick from one player. Skript transparently cancels all events if the first one is cancelled.

There are potential problems with this change:
* Loss of functionality with event-item and 'click with'
  * Event-item gets item in *random* hand
  * 'click with' is now same as 'click holding'
* Two actual clicks within one tick may be sent by Minecraft client, at least in theory
  * Only hacked clients, though?
* Bad connections may split click events from one action over multiple ticks
  * Stable high latency is not an issue, packet loss or fluctuating ping might be

In short, more testing is needed.

---
**Target Minecraft Versions:**  any
**Requirements:**  none
**Related Issues:** #2176

### CODE REVIEW

File: src/main/java/ch/njol/skript/bukkitutil/ClickEventTracker.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.EventException;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.plugin.EventExecutor;
+import org.bukkit.plugin.java.JavaPlugin;
+
+import com.destroystokyo.paper.event.server.ServerTickEndEvent;
```
Comment (JRoy) [inline src/main/java/ch/njol/skript/bukkitutil/ClickEventTracker.java]: This will hard fail on non-paper servers, a class check should be in place

Comment (bensku) [reply src/main/java/ch/njol/skript/bukkitutil/ClickEventTracker.java]: Fixed.

**Comment by bensku** [comment]
I've made changes to track individual ages of events. Also, when possible, interact event pairs are detected.

**Comment by bensku** [comment]
I would like to get this for 2.4, because this is improvement when it comes to *reliability*. It doesn't have to be perfect, just better than what we have now.

**Comment by bensku** [comment]
I'm merging and taking responsibility for hotfixing if need be.
